aws/resource_aws_ec2_traffic_mirror_filter_rule.go

```diff
@@ -0,0 +1,356 @@
+	conn := meta.(*AWSClient).ec2conn
+	ruleId := d.Id()
+
+	input := &ec2.ModifyTrafficMirrorFilterRuleInput{
+		TrafficMirrorFilterRuleId: &ruleId,
+	}
+
+	var removeFields []*string
+	if d.HasChange("protocol") {
+		n := d.Get("protocol")
+		if n == "0" {
+			removeFields = append(removeFields, aws.String(ec2.TrafficMirrorFilterRuleFieldProtocol))
+		} else {
+			input.Protocol = aws.Int64(int64(d.Get("protocol").(int)))
+		}
+	}
+
+	if d.HasChange("description") {
+		n := d.Get("description")
+		if n != "" {
+			input.Description = aws.String(n.(string))
+		} else {
+			removeFields = append(removeFields, aws.String(ec2.TrafficMirrorFilterRuleFieldDescription))
+		}
+	}
+
+	if d.HasChange("destination_cidr_block") {
+		input.DestinationCidrBlock = aws.String(d.Get("destination_cidr_block").(string))
+	}
+
+	if d.HasChange("source_cidr_block") {
+		input.SourceCidrBlock = aws.String(d.Get("source_cidr_block").(string))
+	}
+
+	if d.HasChange("destination_port_range") {
+		v := d.Get("destination_port_range")
+		n := v.([]interface{})
+		if 0 == len(n) {
+			removeFields = append(removeFields, aws.String(ec2.TrafficMirrorFilterRuleFieldDestinationPortRange))
+		} else {
+			//Modify request that adds port range seems to fail if protocol is not set in the request
+			input.Protocol = aws.Int64(int64(d.Get("protocol").(int)))
+			input.DestinationPortRange = buildTrafficMirrorPortRangeRequest(n)
+		}
+	}
+
+	if d.HasChange("source_port_range") {
+		v := d.Get("source_port_range")
+		n := v.([]interface{})
+		if 0 == len(n) {
+			removeFields = append(removeFields, aws.String(ec2.TrafficMirrorFilterRuleFieldSourcePortRange))
+		} else {
+			//Modify request that adds port range seems to fail if protocol is not set in the request
+			input.Protocol = aws.Int64(int64(d.Get("protocol").(int)))
+			input.SourcePortRange = buildTrafficMirrorPortRangeRequest(n)
+		}
+	}
+
+	if d.HasChange("rule_action") {
+		input.RuleAction = aws.String(d.Get("rule_action").(string))
+	}
+
+	if d.HasChange("rule_number") {
+		input.RuleNumber = aws.Int64(int64(d.Get("rule_action").(int)))
+	}
+
+	if d.HasChange("traffic_direction") {
+		input.TrafficDirection = aws.String(d.Get("traffic_direction").(string))
+	}
+
+	if len(removeFields) > 0 {
+		input.SetRemoveFields(removeFields)
+	}
+
+	_, err := conn.ModifyTrafficMirrorFilterRule(input)
+	if err != nil {
+		return fmt.Errorf("Error modifying rule %v", ruleId)
+	}
+
+	return resourceAwsEc2TrafficMirrorFilterRuleRead(d, meta)
+}
+
+func resourceAwsEc2TrafficMirrorFilterRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	ruleId := d.Id()
+	input := &ec2.DeleteTrafficMirrorFilterRuleInput{
+		TrafficMirrorFilterRuleId: &ruleId,
+	}
+
+	_, err := conn.DeleteTrafficMirrorFilterRule(input)
+	if err != nil {
+		return fmt.Errorf("Error deleting traffic mirror filter rule %v", ruleId)
```

Comment:
This is not returning the API/SDK error for operators and code maintainers to troubleshoot the issue:

```suggestion
		return fmt.Errorf("error deleting EC2 Traffic Mirror Filter Rule (%s): %w", ruleId, err)
```